jina/flow/base.py

```diff
@@ -711,6 +711,10 @@ def add(
         if args.grpc_data_requests and args.runtime_cls == 'ZEDRuntime':
             args.runtime_cls = 'GRPCDataRuntime'
 
+            # grpc data runtime does not support sharding at the moment
+            if kwargs.get('shards') is not None:
```

Comment:
I would check that condition outside the if in 711, because it could happen that the runtime_cls is actually not 'ZEDRuntime' but something else

Reply:
good

Reply:
Makes sense, fixed it.